kernel/src/kernel.rs

```diff
@@ -538,11 +548,9 @@ impl Kernel {
             }
 
             // Check if the scheduler wishes to continue running this process.
-            let continue_process = unsafe {
-                resources
-                    .scheduler()
-                    .continue_process(process.processid(), chip)
-            };
+            let continue_process = resources
+                .scheduler()
+                .continue_process(process.processid(), chip);
```

Comment:
It looks like several functions inside the `Scheduler` trait are marked `unsafe` while there is no need. Since no comments are talking about safety, I have assumed that they are artefacts of old code.